src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperContent.cs

```diff
@@ -19,14 +19,9 @@ public abstract class TagHelperContent : IHtmlContentBuilder
         public abstract bool IsModified { get; }
 
         /// <summary>
-        /// Gets a value indicating whether the content is empty.
+        /// Gets a value indicating whether the content is empty or whitespace.
         /// </summary>
-        public abstract bool IsEmpty { get; }
-
-        /// <summary>
-        /// Gets a value indicating whether the content is whitespace.
-        /// </summary>
-        public abstract bool IsWhiteSpace { get; }
+        public abstract bool IsEmptyOrWhiteSpace { get; }
```

Comment:
With this PR the complexity of `DefaultTagHelperContent` increases significantly. That was the initial driving factor in removing the property. @rynowak and I discussed offline about possibly removing `IsEmpty` and `IsWhiteSpace` as part of this PR. I investigated what it would mean to remove the two properties and found that it really only made sense to remove `IsEmpty` and re-purpose the existing `IsWhiteSpace` implementation. I've left a list of reasons at the end of this response.

I'd prefer not to spend time reworking `TagHelperContent.IsEmpty` and its existing tests to undo that work one PR later. I apologize if this seemed off-base @dougbu, @rynowak and I have been working closely on our perf findings/changes in this area which is why I only cc'd him at the beginning of this PR; I've been trying to include others where it makes sense to prevent misunderstandings like this :smile:.

-------------------------------------------------------

Full list of reasons for removing `IsEmpty` as mentioned above:

- We don't use `IsEmpty` in any of our Mvc `TagHelper` implementations.
- We've never seen a user use `IsEmpty` for their own `TagHelper`s.
- `IsEmpty` only makes an extreme edge case scenario more performant. Functionally it can be replaced with `content.GetContent().Length == 0`.
- `IsEmpty` was initially designed with the intent that our Mvc `TagHelper`s would need it; and if they would need it, so would other `TagHelper` author's `TagHelper`s. Ultimately, `IsEmpty` doesn't provide enough value from a maintenance, API or functional perspective to warrant its existence.